osu.Game/Screens/Play/Player.cs

```diff
@@ -345,9 +366,42 @@ private bool onFail()
             if (PauseOverlay.State == Visibility.Visible)
                 PauseOverlay.Hide();
 
+            // Disable input to avoid hitting objects while falling
+            GameplayClockContainer.BlockGameplayInput = true;
+
+            return true;
+        }
+
+        private void updateFail()
+        {
+            audio.Track.Frequency.Value -= Game.IsActive.Value ? 0.0015 : 0.00075;
+
+            GameplayClockContainer.UserPlaybackRate.Value -= Game.IsActive.Value ? 0.0015 : 0.00075;
+
+            DrawableRuleset.Playfield.Alpha -= Game.IsActive.Value ? 0.0015f : 0.00075f;
+
+            Random objRand = new Random();
+            foreach (DrawableHitObject Object in DrawableRuleset.Playfield.AllHitObjects.ToList())
+            {
+                Object.Rotation += (objRand.Next(0, 1) != 0 ? -(float)objRand.NextDouble() : (float)objRand.NextDouble()) / 10;
+                Object.X += (objRand.Next(0, 1) != 0 ? (float)objRand.NextDouble() : -(float)objRand.NextDouble()) / 10;
+                Object.Y += (objRand.Next(0, 1) != 0 ? -(float)objRand.NextDouble() : (float)objRand.NextDouble());
+            }
+
+            if (audio.Track.Frequency.Value > 0)
+                return;
+
+            IsFailing = false; // Stop slowing down the audio
+
+            GameplayClockContainer.Stop();
+
+            audio.Track.Frequency.Value = 1;
```

Comment:
why?

Reply:
Track's playback speed must be restored for other uses

Reply:
What if it wasn't `1` to begin with? Example: playing with Double/Half Time?

Reply:
Wait, Double Time and Half Time doesn't use Frequency afaik, so it'll be 1 in either way + i'm gonna push a huge change in a few minutes